### PR TITLE
Fix persistent secret file permissions

### PR DESCRIPTION
**Pull request by samansmink**
Fixes permissions for the persistent secret files, which are currently too permissive.
Adds tests to confirm that we actually write them with the correct permissions.

### CODE REVIEW

File: src/common/local_file_system.cpp [hunk 1]
```diff
@@ -267,6 +267,37 @@ static string AdditionalProcessInfo(FileSystem &fs, pid_t pid) {
...
 
+void LocalFileSystem::CreatePrivateFile(const string &path_p, FileOpener *opener) {
+	auto path = FileSystem::ExpandPath(path_p, opener);
+
+	int open_flags = O_WRONLY;
+	open_flags |= O_CLOEXEC;
+	open_flags |= O_CREAT;
+	open_flags |= O_APPEND;
+
+	int fd = open(path.c_str(), open_flags, 0600);
```
Comment (Mytherin) [inline src/common/local_file_system.cpp:278]: Maybe we can pass this as a flag to `OpenFile` instead of having this be a separate API call?

Comment (samansmink) [reply src/common/local_file_system.cpp:278]: Yep good idea

**Comment by samansmink** [comment]
thx @Mytherin ! was literally committing the same thing this sec 😅

**Comment by Mytherin** [comment]
Hah, no worries

**Pull request merged**